### PR TITLE
Feat: update asset endpoints

### PR DESCRIPTION
**Pull request by OlfaBensoussia**
## Description
With the EDC milestone 9 release, there were some changes when it comes to the asset related entities. 
This PR addresses all of them but doesn't add the new asset endpoints.
To sum it up, we:
- Added a new Context object
- Updated the CreateResult object
- Fixed the wrong endpoint that was used to get an asset's data address
- Updated Asset objects and added new ones 
- Updated tests
- Updated some utilities.

### How to test it
- Asset related tests should be passing successfully 

---

## Approach
- To understand why those changes had to be made, we checked the [management API](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.0.1-SNAPSHOT#/AssetEntryNewDto) and made use of the open source [POSTMAN collection](https://github.com/eclipse-tractusx/tractusx-edc/blob/2896d385ee9815377461255b58431495796e8fa4/docs/development/postman/collection.json) published by the Tractus-x team
- Those are some screenshots from my local postman to give more context: 
 1. Creating a new asset: **POST** `{{ConnectorManagementURL}}/v2/assets`
Screenshot 2023-05-30 at 6 17 42 PM

2. Getting the list of assets: **POST** `{{ConnectorManagementURL}}/v2/assets/request`
Screenshot 2023-05-30 at 6 17 57 PM

3. Getting an asset: **GET**  `{{ConnectorManagementURL}}/v2/assets/{{AssetId}}`
Screenshot 2023-05-30 at 6 24 01 PM

4. Getting an asset's data address: **GET** `{{ConnectorManagementURL}}/v2/assets/{{AssetId}}/dataaddress`
Screenshot 2023-05-30 at 6 24 35 PM

### Open Questions and Pre-Merge TODOs

- [ ] Update readme examples
- [ ] Check if we have added all of the asset's primitive types or not. We already have `description`, `name` and `contenttype`. I remember we have more than that but I will need to go through the base code to get them.

### CODE REVIEW

File: src/entities/context.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface Context {
+  dct: string;
+  edc: string;
+  dcat: string;
+  odrl: string;
+  dspace: string;
+}
+
+export interface ContextProperties {
+  "@context": Partial<Context>;
```
Comment (fdionisi) [inline src/entities/context.ts:10]: Can a `@context` have more than one field? 

E.g.
```ts
const a: Context = {
  dct: "...",
  edc: "...",
}
```

If not, I don't think `Partial` is the right generic type.

**Review by fdionisi — COMMENTED** [review]
Is there something we should do to enable testing?
Have we considered already using some `json-ld` package to avoid working on the types ourselves?

File: src/entities/context.ts (hunk 1 above)
Comment (fdionisi) [reply src/entities/context.ts:10]: Now I see the response type. Forget my Q.

**Comment by OlfaBensoussia** [comment]
> Is there something we should do to enable testing? Have we considered already using some `json-ld` package to avoid working on the types ourselves?
@fdionisi thank you so much for taking the time to review this.
- By enabling testing, do you mean fixing the failing ones or something else?  
- Before making those changes, I had a look at the `jsonld` package to see if it can be useful to use here. After consideration, it seemed to me that the package is only useful if you want to make some manipulations [compacting, flattening..] on the **json** objects. In our case, I didn't feel like there will an added value for using it, especially that we're not actively making changes on the objects.

**Comment by fdionisi** [comment]
@OlfaBensoussia, sorry for my poor communication. I meant that we should make tests pass, but now we are on the latest version of the connector. We should try land #35 first and then use the docker image to endure correct functionality on this PR. I don't think we'll be developing anymore for milestone-8 (except for #20, for now). Hence, it should be find to proceed

**Comment by fdionisi** [comment]
> * Before making those changes, I had a look at the `jsonld` package to see if it can be useful to use here. After consideration, it seemed to me that the package is only useful if you want to make some manipulations [compacting, flattening..] on the **json** objects. In our case, I didn't feel like there will an added value for using it, especially that we're not actively making changes on the objects.

I hear you. It makes sense to me for now, but I would like to explore the possibility of using a json-ld-aware library to perform input validation - if the format is not correct, then it's not worth bothering the backend in my opinion. Let's keep this exploration for later.

**Comment by OlfaBensoussia** [comment]
@ndr-brt I tried to fix most of the failing tests, remove any hardcoded values and update most of the entities. As far as it concerns the management tests, they are all now passing except the ones that rely on the function `waitForNegotiationState`.
I have a gut feeling that there is an issue with the Negotiation states in general. As an example, canceling a contract negotiation doesn't automatically change the state to **ERROR**, it stays as **TERMINATING** for a while first. 
I still need your help in understanding that part and fixing the remaining failing ones. Once we do that, we should be good to consider that we have the Lib compatible with v0.1+

**Comment by ndr-brt** [comment]
> @ndr-brt I tried to fix most of the failing tests, remove any hardcoded values and update most of the entities. As far as it concerns the management tests, they are all now passing except the ones that rely on the function `waitForNegotiationState`. I have a gut feeling that there is an issue with the Negotiation states in general. As an example, canceling a contract negotiation doesn't automatically change the state to **ERROR**, it stays as **TERMINATING** for a while first. I still need your help in understanding that part and fixing the remaining failing ones. Once we do that, we should be good to consider that we have the Lib compatible with v0.1+

`TERMINATING` it is correct, the [state machine changed a lot since then](https://eclipse-edc.github.io/docs/#/submodule/Connector/docs/developer/architecture/usage-control/contracts?id=state-machine). Please note that the `/cancel` endpoint [has been deprecated as well](https://app.swaggerhub.com/apis/eclipse-edc-bot/management-api/0.2.0#/Contract%20Negotiation/cancelNegotiation), replaced by `/terminate`.

I will take a look at it, great work so far. :raised_hands:

**Comment by ndr-brt** [comment]
@OlfaBensoussia I fixed stuff here and there, the CI is green. I'll propose to merge this into my PR and then I will fixup some cosmetics that needs to be in place before release, let me know.

**Comment by OlfaBensoussia** [comment]
@ndr-brt Thank you for the clarifications. I will merge this and we can fix the rest of the tests on the original branch

**Pull request merged**